packages/auth/amplify_auth_cognito/example/integration_test/utils/setup_utils.dart

```diff
@@ -13,17 +13,26 @@
 // permissions and limitations under the License.
 //
 
+import 'dart:convert';
+
 import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
 import 'package:amplify_auth_cognito_example/amplifyconfiguration.dart';
 import 'package:amplify_flutter/amplify_flutter.dart';
 
-Future<void> configureAuth({
-  List<AmplifyPluginInterface> additionalPlugins = const [],
-}) async {
+Future<void> configureAuth(
+    {List<AmplifyPluginInterface> additionalPlugins = const [],
+    bool customAuth = false}) async {
   if (!Amplify.isConfigured) {
     final authPlugin = AmplifyAuthCognito();
+    String config = amplifyconfig;
```

Comment:
*nit:* Might be nice to move this out to a separate fn.

```dart
String config = _createConfig(amplifyconfig, customAuth: customAuth);
```

```dart
String _createConfig(String amplifyconfig, {bool customAuth = false}) {
  String config = amplifyconfig;
  if (customAuth) {
    var configString = jsonDecode(amplifyconfig);
    configString['auth']['plugins']['awsCognitoAuthPlugin']['Auth']['Default']
        ['authenticationFlowType'] = 'CUSTOM_AUTH';
    config = jsonEncode(configString);
  }
  return config;
}
```

Reply:
I was thinking it might be easier to use AmplifyConfig.fromJson/toJson here but it looks like `AuthConfig ` doesn't have a copyWith method. This is outside of the scope of this PR, but it seems like it would make sense to add to all the plugin configs.